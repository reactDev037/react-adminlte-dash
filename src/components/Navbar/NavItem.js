/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import styled from 'styled-components';

import {
  fontFamilyBase,
  fontSizeBase,
  lineHeightBase,
  fontWeightBase,
  navbarHeight,
  navbarPaddingHorizontal,
  navbarPaddingVertical,
} from '../../styles/variables';

const imageSize = `${Math.floor(parseInt(navbarHeight, 10) / 2)}px`;
const imageMarginTop = `-${Math.ceil(
  ((parseInt(imageSize, 10) +
  parseInt(navbarPaddingHorizontal, 10) +
  parseInt(navbarPaddingVertical, 10)) -
  parseInt(navbarHeight, 10)) / 2)}px`;
const imageMarginBottom = `-${Math.floor(
  ((parseInt(imageSize, 10) +
  parseInt(navbarPaddingHorizontal, 10) +
  parseInt(navbarPaddingVertical, 10)) -
  parseInt(navbarHeight, 10)) / 2)}px`;

const StyledImage = styled.img`
  box-sizing: border-box;
  float: left;
  border: 0;
  vertical-align: middle;
  width: ${imageSize};
  height: ${imageSize};
  border-radius: 50%;
  margin-right: 10px;
  margin-top: ${imageMarginTop};
  margin-bottom: ${imageMarginBottom};
  max-width: none;

  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */
`;

const StyledLink = styled.a`
  color: ${props => props.theme.navbarFontColor || '#fff'};
  text-decoration: none;
  cursor: pointer;
  -webkit-touch-callout: none; /* iOS Safari */
  -webkit-user-select: none; /* Chrome/Safari/Opera */
  -khtml-user-select: none; /* Konqueror */
  -moz-user-select: none; /* Firefox */
  -ms-user-select: none; /* Internet Explorer/Edge */
  user-select: none; /* Non-prefixed version, currently not supported by any browser */
`;

const StyledItem = styled.li`
  /* shared */
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-family: ${fontFamilyBase};
  font-weight: ${fontWeightBase};
  font-size: ${fontSizeBase};
  line-height: ${lineHeightBase};
  box-sizing: border-box;

  float: left;
  background-color: transparent;
  background-image: none;
  border: none;
  outline: none;
  padding: ${navbarPaddingVertical} ${navbarPaddingHorizontal};
  position: relative;
  text-decoration: none;
  cursor: pointer;
  &:focus, &:active { background: transparent; }

  /* theme */
  color: ${props => props.theme.navbarFontColor || '#fff'};
  border-left: ${props => props.theme.navbarItemBorder || 'none'};
  &:hover {
    color: ${props => props.theme.navbarHoverColor || '#fff'};
    background-color: ${props => props.theme.logoBgColor || 'transparent'};
  }
`;

const NavItem = ({ children, onClick, href, image }) => (
  <StyledItem>
    {onClick &&
      <StyledLink onClick={onClick} href="#">
        {image && <StyledImage src={image} />}
        {children}
      </StyledLink>
    }
    {(!onClick && href) &&
      <StyledLink href={href}>
        {image && <StyledImage src={image} />}
        {children}
      </StyledLink>
    }
  </StyledItem>
);

NavItem.propTypes = {
  children: React.PropTypes.node,
  onClick: React.PropTypes.func,
  href: React.PropTypes.string,
  image: React.PropTypes.string,
};

export default NavItem;
