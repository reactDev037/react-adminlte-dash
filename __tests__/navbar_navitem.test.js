/* eslint-env jest */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import red from '../src/styles/skin-red';
import NavItem from '../src/components/Navbar/NavItem';

describe('<NavItem />', () => {
  it('renders correctly with default options', () => {
    const component = renderer.create(<NavItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with theme', () => {
    const component = renderer.create(<ThemeProvider theme={red}>
      <NavItem />
    </ThemeProvider>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with link', () => {
    const component = renderer.create(<NavItem href="/a/link" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with image', () => {
    const component = renderer.create(<NavItem image="/some/path" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
