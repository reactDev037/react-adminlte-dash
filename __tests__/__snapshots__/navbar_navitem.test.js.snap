exports[`<NavItem /> renders correctly with default options 1`] = `
<li
  className="dzjnOV" />
`;

exports[`<NavItem /> renders correctly with image 1`] = `
<li
  className="dzjnOV" />
`;

exports[`<NavItem /> renders correctly with link 1`] = `
<li
  className="dzjnOV">
  <a
    className="dfofQy"
    href="/a/link" />
</li>
`;

exports[`<NavItem /> renders correctly with theme 1`] = `
<li
  className="gDCOpn" />
`;
